Sources/web3swift/Web3/Web3+GasOracle.swift

```diff
@@ -0,0 +1,214 @@
+//
+//  Web3+GasOracle.swift
+//  web3swift
+//
+//  Created by Yaroslav on 31.03.2022.
+//  Copyright © 2022 web3swift. All rights reserved.
+//
+
+import Foundation
+import BigInt
+
+extension Web3 {
+    /// Oracle is the class to of a transaction fee suggestion
+    ///
+    /// It designed for EIP-1559 transactions only.
+    final public class Oracle {
+        private var latestBlock: Block?
+
+        /// Web3 provider by wich accessing to the blockchain
+        private let web3Provider: web3
+
+        /// Ethereum scope shortcut
+        private var eth: web3.Eth { web3Provider.eth }
+
+        /// Number of block to caltulate statistics
+        public private(set) var blocksNumber: BigUInt
+
+        /// Number of transacrtions to filter block for tip calculation
+        public private(set) var transactionsNumber: BigUInt
+
+        /// Oracle initializer
+        /// - Parameters:
+        ///   - provider: Web3 Ethereum provider
+        ///   - blocksNumber: Number of block to caltulate statistics
+        ///   - transactionsNumber: Number of transacrtions to filter block for tip calculation
+        public init(_ provider: web3, blocksNumber: BigUInt = 20, transactionsNumber: BigUInt = 50) {
+            self.web3Provider = provider
+            self.blocksNumber = blocksNumber
+            self.transactionsNumber = transactionsNumber
+        }
+
+        private func calcBaseFee(for block: Block?) -> BigUInt {
+            guard let block = block else { return 0 }
+            return Web3.calcBaseFee(block)
+        }
+
+        private func calculateStatistic(_ data: [BigUInt], _ statistic: Statistic) throws -> BigUInt {
```

Comment:
can we use Swift Styled labeled parameters ie  calculateStatistic(x,y) should have named parameters

Reply:
Could you please provide any desired method signature?

Also please note that it's a private method and it shouldn't be used by any other class.

Reply:
I think he's suggesting to not have the parameters be anonymous in the call. 
So instead of:
`private func calculateStatistic(_ data: [BigUInt], _ statistic: Statistic) throws -> BigUInt`
you have:
`private func calculateStatistic(data: [BigUInt],  statistic: Statistic) throws -> BigUInt`

meaning that when you call instead of `calculateStatistic(x, y)` you would have `calculateStatistic(data: x, statistic: y)` 

My suggestion would be to swap the parameters as well and name the first one`for:` giving you
`private func calculateStatistic(for statistic: Statistic, data: [BigUInt]) throws -> BigUInt`
so your call would look like
`calculateStatistic(for: .mean, data: data)`

Even though it is a private function, improving the clarity in the call would help anyone looking at the code down the road.